### ISSUE TITLE
u3d/install: some versions of unity not found

### ISSUE THREAD

**Issue by marklonquist** [DragonBox/u3d#172]
Hi

I've had a few issues where u3d can't find the version of the project in the list of available versions.

This results in time spent on manual installation 😢 

Latest issue with; **5.6.3p4**. Also happened with version **2017.1.0p4**.

**Comment by niezbop** [comment]
Hey @marklonquist !

I just took a look at your issue and I understand where it comes from, we only look for the patches on the first page of the patches list of unity, and the two versions you are talking about are on other pages!

Right now, I'm not sure I can fix it:
- No time on my side
- I'm not sure we want to check ~40 pages of patches every time we run `u3d available`, as it would cause quite a performance loss.
- We're planning a service to provide a centralized cache for every version, so it would solve the performance issue mentionned before, but we first need to do this.

Nonetheless, if you want to try to tackle it yourself, you could fiddle with [this](https://github.com/DragonBox/u3d/blob/master/lib/u3d/unity_versions.rb#L215) and iterate over the pages (at https://unity3d.com/fr/unity/qa/patch-releases?page=`[your iterator]`) to retrieve every patch!

**Comment by marklonquist** [comment]
Thanks @niezbop I'll dig around :-)

**Comment by lacostej** [comment]
Another thing to take into account, we need to filter versions < 5 from the list of patch versions.

**Comment by marklonquist** [comment]
Just a quick thought;

When running the install command, wouldn't it be possible to simple check for the version using a more direct method?

If version contains "p", check if it exist by simply using the version in this URL:
https://unity3d.com/unity/qa/patch-releases/<VERSION> (ex.: 2017.1.1p4)

**Comment by lacostej** [comment]
I would prefer if we fill the cache properly and not change the logic too much.

We have a task to move the generation of the cache online (#142). That way u3d will download_cache and then install.

**Comment by marklonquist** [comment]
yea that would fix the issue. I'll try to make a hacky fix for my own setup until #142 is implemented.

**Comment by lacostej** [comment]
@marklonquist here's something that might help

````
U3d::Utils.get_ssl("https://unity3d.com/unity/qa/patch-releases").scan(/\/unity\/qa\/patch-releases\?page=\d+/)
````

**Comment by lacostej** [comment]
@marklonquist solved in 1.0.9, freshly released.

**Comment by lacostej** [comment]
@marklonquist you will be happy to know that u3d available is almost instantaneous thanks to the u3d available improvements in 1.0.13. We fetch the information from a centralized cache.

**Comment by marklonquist** [comment]
Sounds awesome :) Will update our building machine. Thanks for letting me know.
